pandas/core/generic.py

```diff
@@ -4850,7 +4850,7 @@ def _where(self, cond, other=np.nan, inplace=False, axis=None, level=None,
 
         msg = "Boolean array expected for the condition, not {dtype}"
 
-        if not isinstance(cond, pd.DataFrame):
+        if not isinstance(cond, (pd.DataFrame, pd.Panel)):
```

Comment:
I see that this code is relatively new (not in 0.19.2) 
Can you explain why my adaptation is wrong ? the cond argument can be a Panel, for instance when calling Panel.mask(cond=some other panel), and if you do not adapt the test as I propose, it will run the two following lines that uses `cond.dtype` that does not exist for Panel (test fails)